dissect/target/plugins/apps/ssh/sshd.py

```diff
@@ -0,0 +1,179 @@
+from typing import TYPE_CHECKING, Any, Callable, Iterator, Optional, Union
+
+from dissect.target import Target
+from dissect.target.exceptions import UnsupportedPluginError
+from dissect.target.helpers.record import DynamicDescriptor, TargetRecordDescriptor
+from dissect.target.plugin import Plugin, export
+from dissect.target.plugins.apps.ssh.openssh import CommonSSHD
+
+if TYPE_CHECKING:
+    from dissect.target.plugins.general.config import ConfigurationTreePlugin
+
+SSHD_BOOLEAN_VALUES = (
+    "yes",
+    "no",
+)
+
+SSHD_BOOLEAN_FIELDS = (
+    "AllowAgentForwarding",
+    "DebianBanner",
+    "DisableForwarding",
+    "ChallengeResponseAuthentication",
+    "ExposeAuthInfo",
+    "GSSAPIAuthentication",
+    "GSSAPICleanupCredentials",
+    "GSSAPICleanupCredentials",
+    "GSSAPIStrictAcceptorCheck",
+    "GSSAPIStoreCredentialsOnRekey",
+    "GSSAPIKeyExchange",
+    "GSSAPIStrictAcceptorCheck",
+    "GSSAPIStoreCredentialsOnRekey",
+    "HostbasedAuthentication",
+    "HostbasedUsesNameFromPacketOnly",
+    "IgnoreUserKnownHosts",
+    "KbdInteractiveAuthentication",
+    "KeepAlive",
+    "KerberosAuthentication",
+    "KerberosGetAFSToken",
+    "KerberosOrLocalPasswd",
+    "KerberosTicketCleanup",
+    "PasswordAuthentication",
+    "PermitEmptyPasswords",
+    "PermitTTY",
+    "PermitUserRC",
+    "PrintLastLog",
+    "PrintMotd",
+    "PubkeyAuthentication",
+    "StreamLocalBindUnlink",
+    "StrictModes",
+    "TCPKeepAlive",
+    "UsePAM",
+    "X11Forwarding",
+    "X11UseLocalhost",
+)
+
+SSHD_INTEGER_FIELDS = (
+    "ClientAliveCountMax",
+    "ClientAliveInterval",
+    "LoginGraceTime",
+    "MaxAuthTries",
+    "MaxSessions",
+    "Port",
+    "X11DisplayOffset",
+)
+
+SSHD_MULTIPLE_DEFINITIONS_ALLOWED_FIELDS = (
+    "AcceptEnv",
+    "Include",
+    "ListenAddress",
+    "PermitListen",
+    "PermitOpen",
+    "Port",
+)
+
+
+class SSHServerPlugin(Plugin):
+    __namespace__ = "sshd"
```

Comment:
Would it make sense to put the public/private keys/etc functions of the `ssh` plugin for the server directory in this plugin? @JSCU-CNI?

As also mentioned in another PR, that would probably mean moving the current `ssh` plugin to `openssh` and creating a namespaced `ssh` plugin that would wrap `openssh`, `putty`, and probably `dropbear` down the line.

Reply:
As discussed, this approach would make sense in our opinion.